hooks/command

```diff
@@ -221,34 +172,49 @@ echo "--- Getting the password to unlock the keychain"
 keychain_pw_helper_script=$(find_keychain_pw_helper_script)
 keychain_pw=$(get_keychain_pw "${keychain_pw_helper_script}")
 
-echo "--- Unlocking the keychain"
+#echo "--- Unlocking the keychain"
 unlock_keychain "${codesigning_keychain}" "${keychain_pw}"
 
 echo "--- Setting up local directories"
 # Download things in a local dir to identify them
 artifact_dir_fragment="artifacts"
 rm -rf "${artifact_dir_fragment}" && mkdir -p "${artifact_dir_fragment}"
-relative_artifacts_dir="$(pwd)/${artifact_dir_fragment}"
-
-# Signed things go in a local dir so the uploaded artifacts don't have a weird path
-signed_dir_fragment="signed"
-rm -rf "${signed_dir_fragment}" && mkdir -p "${signed_dir_fragment}"
-
-# Download and sign each requested artifact
-for artifact in $(plugin_read_list INPUT_ARTIFACTS); do
-  echo "--- Signing ${artifact}"
+absolute_artifacts_dir="$(pwd)/${artifact_dir_fragment}"
 
+## Download and sign each requested artifact
+plugin_read_list INPUT_ARTIFACTS | while read -r artifact; do
   echo "${artifact}: retrieving unsigned artifact"
-  unsigned_artifact="$(fetch_artifact "${artifact}" "${relative_artifacts_dir}")"
-
-  sign_and_notarize "${unsigned_artifact}" "${signed_dir_fragment}"
-  retval=$?
-  if [[ "${retval}" -ne 0 ]]; then
-    echo "failed to sign and notarize '${unsigned_artifact}': error code '${retval}'"
-    exit ${retval}
-  fi
-
+  fetch_artifact "${artifact}" "${absolute_artifacts_dir}"
 done
 
-echo "--- Uploading artifacts"
-upload_artifact "${signed_dir_fragment}/**/*"
+entitlements="${BUILDKITE_PLUGIN_MAC_CODESIGN_ENTITLEMENTS}"
+if [[ -n "${entitlements}" ]]; then
+  #  fetch_artifact "${entitlements}" "${absolute_artifacts_dir}"
+  entitlements="${absolute_artifacts_dir}/${entitlements}"
+fi
+
+config_path="config.json"
+prerequisites="$(plugin_read_list SIGN_PREREQUISITES)"
+
+# generate config
+echo "-- Generating signing config"
+generate_signing_config "${prerequisites}" "${entitlements}" "${absolute_artifacts_dir}" "${config_path}"
+
+echo "-- Executing gon on generated config"
+gon "${config_path}"
+retval=$?
+if [[ "${retval}" -ne 0 ]]; then
+  echo "notarization failed: error code '${retval}'"
+  exit 5
+fi
+
+echo "--- Uploading output artifacts"
+# upload config
+echo "${config_path}: uploading config used"
+upload_artifact "${config_path}"
+
+# upload signed items
+plugin_read_list OUTPUT_ARTIFACTS | while read -r artifact; do
```

Comment:
Seconding the earlier question about whether this is necessary anymore, if we're limiting to one signed target.